src/main/java/net/minestom/server/extras/bungee/BungeeCordProxy.java

```diff
@@ -56,4 +87,47 @@ public static PlayerSkin readSkin(@NotNull String json) {
             return null;
         }
     }
+
+    public static @NotNull Pair<PlayerSkin, Component> readSkinBungee(@NotNull String json) {
+        JsonArray array = JsonParser.parseString(json).getAsJsonArray();
+        boolean foundToken = false;
+        Component message = NO_BUNGEE_GUARD_TOKEN;
+        PlayerSkin skin = null;
+
+        for (JsonElement element : array) {
+            JsonObject jsonObject = element.getAsJsonObject();
+            JsonElement name = jsonObject.get("name");
+            if (name != null) {
+                switch (name.getAsString()) {
+                    case "textures" -> {
+                        JsonElement value = jsonObject.get("value");
+                        JsonElement signature = jsonObject.get("signature");
+                        if (value == null || signature == null) continue;
+
+                        skin = new PlayerSkin(value.getAsString(), signature.getAsString());
+                    }
+                    case "bungeeguard-token" -> {
+                        if (foundToken) {
+                            message = MULTIPLE_TOKENS;
+                            continue;
```

Comment:
seems odd to continue here, isn't multiple tokens an error? same with continuing when value == null and !bungeeGuardTokens.contains(value.getAsString())

Reply:
I just did it to continue parsing the PlayerSkin, should I just stop in general?

Reply:
i think stopping makes more sense, but is this code based on an implementation you found elsewhere?

Reply:
no